packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -390,10 +390,40 @@ export const toStyles = ( tree, blockSelectors, hasBlockGapSupport ) => {
 
 		// Process the remaning block styles (they use either normal block class or __experimentalSelector).
 		const declarations = getStylesDeclarations( styles );
-		if ( declarations.length === 0 ) {
-			return;
+		if ( declarations?.length ) {
+			ruleset = ruleset + `${ selector }{${ declarations.join( ';' ) };}`;
+		}
+
+		// Check for pseudo selector in `styles` and handle separately.
+		const psuedoSelectorStyles = Object.entries( styles ).filter(
+			( [ key ] ) => key.startsWith( ':' )
+		);
+
+		if ( psuedoSelectorStyles?.length ) {
```

Comment:
Code style nit:

```suggestion
		if ( ! psuedoSelectorStyles?.length ) {
			return
		}
```

Less nesting, less thinking required.

Reply:
I haven't looked in detail but do we definitely want to exit the function here?

Reply:
We're inside of a `forEach` here and this is the last operation so I'd say it's fine to exit. If we want to have multiple segments, we could call `forEach` multiple times like `.forEach(/* do x */).forEach(/* do y */)`. instead of making it one mega function with `if(x) { /* lots of code */ } if(y) { /* lots of code */ }`. But that's beyond the scope of this PR, feel free to make the call on whether you'd like to keep the nesting or not – I'm just expressing opinions, not blockers :-)

Reply:
I looked at this and realised we're in the loop (like you said). All good. I agree let's exit early.